op-e2e/sgt/sgt_test.go

```diff
@@ -0,0 +1,389 @@
+	// ensure that native balance is 0
+	require.Equal(t, common.Big0.Cmp(l2Value), 0)
+	// ensure that txValue is 0
+	require.Equal(t, common.Big0.Cmp(txValue), 0)
+	testAccount, _ := setUpTestAccount(t, ctx, index, sgt, sgtValue, l2Value)
+
+	// make a simple tx with the testAccount: transfer txValue from testAccount to dummyAddr
+	_, err := sgt.transferNativeToken(t, ctx, testAccount, dummyAddr, txValue)
+	require.Error(t, err)
+}
+
+func fullNativeInsufficientGasPaymentFail(t *testing.T, ctx context.Context, index int64, sgtValue *big.Int, l2Value *big.Int, txValue *big.Int, sgt *SgtHelper) {
+	// ensure that sgt balance is 0
+	require.Equal(t, common.Big0.Cmp(sgtValue), 0)
+	// ensure that txValue is 0
+	require.Equal(t, common.Big0.Cmp(txValue), 0)
+	testAccount, _ := setUpTestAccount(t, ctx, index, sgt, sgtValue, l2Value)
+
+	// make a simple tx with the testAccount: transfer txValue from testAccount to dummyAddr
+	_, err := sgt.transferNativeToken(t, ctx, testAccount, dummyAddr, txValue)
+	require.Error(t, err)
+}
+
+func partialSGTInsufficientGasPaymentFail(t *testing.T, ctx context.Context, index int64, sgtValue *big.Int, l2Value *big.Int, txValue *big.Int, sgt *SgtHelper) {
+	// ensure that txValue is 0
+	require.Equal(t, common.Big0.Cmp(txValue), 0)
+	testAccount, _ := setUpTestAccount(t, ctx, index, sgt, sgtValue, l2Value)
+
+	// make a simple tx with the testAccount: transfer txValue from testAccount to dummyAddr
+	_, err := sgt.transferNativeToken(t, ctx, testAccount, dummyAddr, txValue)
+	require.Error(t, err)
+}
+
+func fullSGTGasPaymentAndNonZeroTxValueWithInsufficientNativeBalanceFail(t *testing.T, ctx context.Context, index int64, sgtValue *big.Int, l2Value *big.Int, txValue *big.Int, sgt *SgtHelper) {
+	// ensure that native balance is less than txValue
+	require.Equal(t, l2Value.Cmp(txValue), -1)
+	testAccount, _ := setUpTestAccount(t, ctx, index, sgt, sgtValue, l2Value)
+
+	// make a simple tx with the testAccount: transfer txValue from testAccount to dummyAddr
+	tx, err := sgt.transferNativeToken(t, ctx, testAccount, dummyAddr, txValue)
+	// ensure sgt balance is large enough to cover the gas cost
+	require.Equal(t, sgtValue.Cmp(tx.GasCost()), 1)
+	require.Error(t, err)
+}
+
+func partialSGTGasPaymentAndNonZeroTxValueWithInsufficientNativeBalanceFail(t *testing.T, ctx context.Context, index int64, sgtValue *big.Int, l2Value *big.Int, txValue *big.Int, sgt *SgtHelper) {
+	testAccount, _ := setUpTestAccount(t, ctx, index, sgt, sgtValue, l2Value)
+
+	// make a simple tx with the testAccount: transfer txValue from testAccount to dummyAddr
+	tx, err := sgt.transferNativeToken(t, ctx, testAccount, dummyAddr, txValue)
+	// ensure native balance is large enough to cover the gas cost
+	require.Equal(t, l2Value.Cmp(tx.GasCost()), 1)
+	require.Error(t, err)
+}
+
+func calcGasFee(receipt *types.Receipt) *big.Int {
```

Comment:
Where are the tests that check whether the balances of the vaults (fee recipients) are expected

Reply:
It'll be included in the next PR